src/transaction.js

```diff
@@ -304,6 +304,9 @@ Transaction.prototype.toBuffer = function () {
 
   writeUInt32(this.locktime)
 
+  // avoid slicing unless necessary
+  if (initialOffset !== undefined) return buffer.slice(initialOffset, offset)
```

Comment:
this will help in node environments,  but in the browser,  will this still result in a copy anyway?